WeatherTodayTests/Sources/Specs/WeatherViewModelSpec.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  WeatherViewModelSpec.swift
+//  WeatherToday
+//
+//  Created by Taher on 20/10/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Quick
+import Nimble
+import RxCocoa
+import RxSwift
+import RxTest
+import RxNimble
+
+@testable import WeatherToday
+
+final class WeatherViewModelSpec: QuickSpec {
+
+    override func spec() {
+
+        var viewModel: WeatherViewModel!
+        var useCase: MockWeatherUseCase!
+        var disposeBag: DisposeBag!
+        var temperatureText = ""
+        let defaultTemperatureText = "Temperature _ ℃"
+
+        describe("a WeatherViewModel") {
+
+            beforeEach {
+                useCase = MockWeatherUseCase()
+                viewModel = WeatherViewModel(weatherUseCase: useCase)
+                disposeBag = DisposeBag()
+                viewModel.output.temperature.asObservable().subscribe(
+                    onNext: { data in
+                        temperatureText = data
+                    },
+                    onError: { err in
+                        temperatureText = defaultTemperatureText
+                    }
+                )
+                .disposed(by: disposeBag)
+            }
+
+            context("when city name is not provided") {
+
+                beforeEach {
+                    viewModel.input.viewDidLoad()
+                }
+
+                it("should show default text") {
+                    expect(temperatureText) == defaultTemperatureText
+                }
+            }
+
+            context("when valid city name is provided") {
+
+                beforeEach {
+                    viewModel.input.didFinishTyping(cityName: "Dhaka")
+                }
+
+                it("should show temperature text") {
+                    expect(temperatureText) == "Temperature 30.0 ℃"
+                }
+            }
+
+            context("when invalid city name is provided") {
+
+                beforeEach {
+                    viewModel.input.didFinishTyping(cityName: "A")
+                }
+
+                it("should show default text") {
+                    expect(temperatureText) == defaultTemperatureText
+                }
+            }
```

Comment:
I suggest to wrap this block inside another describe for better test structure.
```
describe("its didFinishTyping") {
// this block
}
```

Reply:
fixed in this [commit](https://github.com/nimblehq/ci-cd-sample-ios/pull/22/commits/3d65e1179992cc9c7442660fdb88d9d668fb923c)